internal/cephfs/volume.go

```diff
@@ -157,32 +159,18 @@ func createVolume(ctx context.Context, volOptions *volumeOptions, cr *util.Crede
 		clusterAdditionalInfo[volOptions.ClusterID].subVolumeGroupCreated = true
 	}
 
-	args := []string{
-		"fs",
-		"subvolume",
-		"create",
-		volOptions.FsName,
-		string(volID),
-		strconv.FormatInt(bytesQuota, 10),
-		"--group_name",
-		volOptions.SubvolumeGroup,
-		"--mode", "777",
-		"-m", volOptions.Monitors,
-		"-c", util.CephConfigPath,
-		"-n", cephEntityClientPrefix + cr.ID,
-		"--keyfile=" + cr.KeyFile,
+	opts := fsAdmin.SubVolumeOptions{
+		Size: fsAdmin.ByteCount(bytesQuota),
+		Mode: modeAllRWX,
 	}
-
 	if volOptions.Pool != "" {
-		args = append(args, "--pool_layout", volOptions.Pool)
+		opts.PoolLayout = volOptions.Pool
 	}
 
-	err := execCommandErr(
-		ctx,
-		"ceph",
-		args[:]...)
+	// FIXME: check if the right credentials are used ("-n", cephEntityClientPrefix + cr.ID)
+	err = ca.CreateSubVolume(volOptions.FsName, volOptions.SubvolumeGroup, string(volID), &opts)
 	if err != nil {
-		util.ErrorLog(ctx, "failed to create subvolume %s(%s) in fs %s", string(volID), err, volOptions.FsName)
+		util.ErrorLog(ctx, "failed to create subvolume %s in fs %s: %s", string(volID), volOptions.FsName, err)
```

Comment:
```suggestion
		util.ErrorLog(ctx, "failed to create subvolume %s in fs %s: %v", string(volID), volOptions.FsName, err)
```

Reply:
not an objection, but I'd like to know why, what's the difference?

Reply:
`%s` will do too. It's just that `%v` is the default format for representation of errors and a similar convention is being used throughout the repo. Just a matter of choice here :)

Reply:
In this particular file, all errors are logged with `%s`, so I'll keep it the way it is.

Reply:
actually we any way have it in string so `%s` is fine